build_runner/lib/src/generate/phase.dart

```diff
@@ -5,15 +5,17 @@
 import 'package:build/build.dart';
 import 'package:build_config/build_config.dart';
 import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
 
 import 'input_matcher.dart';
 
 /// A "phase" in the build graph, which represents running a [Builder] on a
 /// specific [package].
-class BuildAction implements InputMatcher {
+class BuildAction {
   final String package;
   final Builder builder;
-  final InputMatcher _inputs;
+  final InputMatcher targetSources;
```

Comment:
Should we consider passing a full TargetNode here?

Reply:
We could, I think if we long-term want to store target information on the AssetGraph and use that for visibility we'd want to...

For now I'm in favor of leaving this as is and waiting for the use case to surface.